src/Features/Core/Portable/CodeStyle/AbstractCodeStyleProvider.Refactoring.cs

```diff
@@ -0,0 +1,86 @@
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    // This part contains all the logic for hooking up the CodeRefactoring to the CodeStyleProvider.
+    // All the code in this part is an implementation detail and is intentionally private so that
+    // subclasses cannot change anything.  All code relevant to subclasses relating to refactorings
+    // is contained in AbstractCodeStyleProvider.cs
+
+    internal abstract partial class AbstractCodeStyleProvider<TOptionKind, TCodeStyleProvider>
+    {
+        private async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var (document, _, cancellationToken) = context;
+
+            var optionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var optionValue = optionSet.GetOption(_option);
+
+            var severity = GetOptionSeverity(optionValue);
+            switch (severity)
+            {
+                case ReportDiagnostic.Suppress:
+                case ReportDiagnostic.Hidden:
+                    // if the severity is Hidden that's equivalent to 'refactoring only', so we want
+                    // to try to compute the refactoring here.
+                    //
+                    // If the severity is 'suppress', that means the user doesn't want the actual
+                    // analyzer to run here.  However, we can still check to see if we could offer
+                    // the feature here as a refactoring.
+                    await ComputeRefactoringsAsync(context, optionValue.Value, analyzerActive: false).ConfigureAwait(false);
+                    return;
+
+                case ReportDiagnostic.Error:
+                case ReportDiagnostic.Warn:
+                case ReportDiagnostic.Info:
+                    // User has this option set at a level where we want it checked by the
+                    // DiagnosticAnalyser and not the CodeRefactoringProvider.  However, we still
+                    // want to check if we want to offer the *reverse* refactoring here in this
+                    // single location.
+                    //
+                    // For example, say this is the "use expression body" feature.  If the user says
+                    // they always prefer expression-bodies (with warning level), then we want the
+                    // analyzer to always be checking for that.  However, we still want to offer the
+                    // refactoring to flip their code to use a block body here, just in case that
+                    // was something they wanted to do as a one off (i.e. before adding new
+                    // statements.
+                    //
+                    // TODO(cyrusn): Should we only do this for warn/info?  Argument could be made
```

Comment:
>TODO(cyrusn) [](start = 23, length = 12)

GitHub issue for this?

Reply:
i'm ok without an issue.  the point of the comment is to at least explain the thinking.  note that this *is* our current behavior, so i'm not changing anything.